### PR TITLE
internal: Performance optimizations

### PR DESCRIPTION
**Pull request by mo8it**
- Use `Command::arg` directly
- Avoid the overhead of the `select!` macro when possible
- Use `select_biased!`

### CODE REVIEW

File: crates/rust-analyzer/src/flycheck.rs [hunk 1]
```diff
@@ -419,7 +420,8 @@ impl FlycheckActor {
                 cmd.arg("--keep-going");
 
                 options.apply_on_command(&mut cmd);
-                (cmd, options.extra_args.clone())
+                cmd.args(&options.extra_args);
```
Comment (mo8it) [inline crates/rust-analyzer/src/flycheck.rs:423]: It is weird that `extra_args` aren't applied in `apply_on_command`

File: crates/vfs-notify/src/lib.rs [hunk 2]
```diff
@@ -85,10 +85,13 @@ impl NotifyActor {
...
     fn next_event(&self, receiver: &Receiver<Message>) -> Option<Event> {
-        let watcher_receiver = self.watcher.as_ref().map(|(_, receiver)| receiver);
+        let Some((_, watcher_receiver)) = &self.watcher else {
+            return receiver.recv().ok().map(Event::Message);
+        };
+
         select! {
             recv(receiver) -> it => it.ok().map(Event::Message),
-            recv(watcher_receiver.unwrap_or(&never())) -> it => Some(Event::NotifyEvent(it.unwrap())),
+            recv(watcher_receiver) -> it => Some(Event::NotifyEvent(it.unwrap())),
```
Comment (mo8it) [inline crates/vfs-notify/src/lib.rs:94]: Why are we unwrapping here but not with other messages?

Comment (Veykril) [reply crates/vfs-notify/src/lib.rs:94]: Unsure, might've been due to drop order at some point to prevent a panic

Comment (mo8it) [reply crates/vfs-notify/src/lib.rs:94]: Well, not relevant for the PR. just noticed it.

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Pull request merged**